components/cam/src/physics/rrtmgp/cam_optics.F90

```diff
@@ -356,347 +338,138 @@ end subroutine combine_properties
-                  optics_out%g(iday,ilev_rad,igpt) = optics_cam%assymmetry_parameter(icol,ilev_cam,iband)
-               else
-                  optics_out%tau(iday,ilev_rad,igpt) = 0._r8
-                  optics_out%ssa(iday,ilev_rad,igpt) = 1._r8
-                  optics_out%g(iday,ilev_rad,igpt) = 0._r8
-               end if
-            end do  ! igpt
-         end do  ! iday
-      end do  ! ilev_cam
-
-      ! Apply delta scaling to account for forward-scattering
-      ! TODO: delta_scale takes the forward scattering fraction as an optional
-      ! parameter. In the current cloud optics_sw scheme, forward scattering is taken
-      ! just as g^2, which delta_scale assumes if forward scattering fraction is
-      ! omitted in the function call. In the future, we should explicitly pass
-      ! this. This just requires modifying the get_cloud_optics_sw procedures to also
-      ! pass the foward scattering fraction that the CAM cloud optics_sw assumes.
-      call handle_error(optics_out%delta_scale())
-
-      ! Check cloud optics_sw
-      call handle_error(optics_out%validate())
-
-      call optics_cam%finalize()
-
-      deallocate(iscloudy)
-
-   end subroutine set_cloud_optics_sw
+   end subroutine compress_optics_sw
 
    !----------------------------------------------------------------------------
 
-   subroutine set_cloud_optics_lw(state, pbuf, kdist, optics_out)
-      
-      use ppgrid, only: pcols, pver
-      use physics_types, only: physics_state
-      use physics_buffer, only: physics_buffer_desc, &
-                                pbuf_get_field, pbuf_get_index
-      use mo_optical_props, only: ty_optical_props_1scl
-      use mo_gas_optics_rrtmgp, only: ty_gas_optics_rrtmgp
+   ! Do MCICA sampling of optics here. This will map bands to gpoints,
+   ! while doing stochastic sampling of cloud state
+   subroutine sample_cloud_optics_lw( &
+         ncol, nlev, ngpt, gpt2bnd, &
+         pmid, cld, cldfsnow, &
+         tau_bnd, tau_gpt)
+      use ppgrid, only: pcols
       use mcica_subcol_gen, only: mcica_subcol_mask
 
-      type(physics_state), intent(in) :: state
-      type(physics_buffer_desc), pointer :: pbuf(:)
-      type(ty_gas_optics_rrtmgp), intent(in) :: kdist
-      type(ty_optical_props_1scl), intent(inout) :: optics_out
-
-      type(cam_optics_type) :: optics_cam
-      real(r8), pointer :: cloud_fraction(:,:)
-      real(r8), pointer :: snow_fraction(:,:)
-      real(r8) :: combined_cloud_fraction(pcols,pver)
+      integer, intent(in) :: ncol, nlev, ngpt
+      integer, intent(in), dimension(:) :: gpt2bnd
+      real(r8), intent(in), dimension(:,:) :: pmid, cld, cldfsnow
+      real(r8), intent(in), dimension(:,:,:) :: tau_bnd
+      real(r8), intent(out), dimension(:,:,:) :: tau_gpt
+      real(r8) :: combined_cld(ncol,nlev)
+      logical :: iscloudy(ngpt,ncol,nlev)
 
       ! For MCICA sampling routine
       integer, parameter :: changeseed = 1
```

Comment:
Might be worth using the comment from the sw version of this routine:
"how many times to permute random seed"
Seems to better describe what this variable does.

Reply:
Done.